### DISCUSSION TITLE
Can we connect locally to a Goodwe Homekit HK1000?

### DISCUSSION THREAD

**Discussion by rvilla87** [mletenay/home-assistant-goodwe-inverter#113] [Q&A]
Hello,

first of all, thanks for your work, it's a really usefull home assistant integration! Thanks to this integration I can get all the data from my inverter without the need of semsportal.

The problem is that I also use a [Goodwe HomeKit 1000](https://en.goodwe.com/single-phase-homekit) and is the one who can give me data about the electrical load of my house.

I can connect locally to the webserver of my HomeKit 1000 but I have no idea how I can get the energy data. I suppose that maybe with the same method as this integration uses but changing some code we can access also to the energy data from the HomeKit 1000.

I tried to use the same method as [described here]([url](https://pypi.org/project/goodwe/)) with no luck because it returns the error: `goodwe.exceptions.InverterError: Unable to connect to the inverter at host=192.xxxxxx, or your inverter is not supported yet.` [...] `RequestRejectedException('ILLEGAL DATA ADDRESS')`.

Is there any way I can do some testing/debugging to the Homekit in order to get the energy information?

Thanks!

**Comment by icepicknz** [comment]
I too am looking to connect the HK1000

**Reply by SniperHilton** [reply]
Isn't the hk1000 just the meter?

**Reply by rvilla87** [reply]
> Isn't the hk1000 just the meter?

The meter and also the one who monitor the load energy consumption. As stated in [the website](https://en.goodwe.com/single-phase-homekit): "_GoodWe HomeKit is a solution designed to monitor load energy consumption in real time, 24 hours a day._"

The thing is, with [this integration](https://github.com/mletenay/home-assistant-goodwe-inverter) we can monitor the generated energy (by our photovoltaic panels) in real time but we can't monitor the load energy (the energy that your house uses). That information is not reported by the inversor (in my case I have a GW 5000D) but by the HomeKit 1000.

**Comment by SniperHilton** [comment]
But the meter reports to the inverter, and we gather the data from the inverter.

Is the HK1000 connected to the network? Mine is connected directly onto the inverter and isn't accessable on its own.

**Reply by rvilla87** [reply]
> But the meter reports to the inverter, and we gather the data from the inverter.
> 
> Is the HK1000 connected to the network? Mine is connected directly onto the inverter and isn't accessable on its own.

Not accessible on your own or maybe you don't know how to access?

The meter reports to the inverter? I think it's just the opposite. The HK1000 (meter) should be independent of the inverter. Watch this video to see how the HK100 works: https://www.youtube.com/watch?v=jFHgczEY2L8

As you can see at minute 1:15, the connection to the inverter is **optional**.

My installation is the same as you can see in the video (also, HK1000 has Wifi connection in order to connect so semsportal). Maybe your installation is different. I know other cases with different inverters where the homekit is "inside" the inverter and they can get all the information at the same time through the inverter. That is not the case of the installation seen in the video (maybe it's a GW 5000D or similar).

**Reply by SniperHilton** [reply]
Yes my installation is different.
Meter is linked to the inverter only. Not connected on WiFi.

The meter then sends the data to the inverter.

This integrateion is for goodwe **inverters** 

You'll need to look for another integration for the HK1000 meter separately. Or link the meter to the inverter and obtain the information from there.

**Reply by rvilla87** [reply]
Yeah, I know this is related to inverters but as both (inverter and smart meter) are from Goodwe maybe the communication is similar for both cases.

For some Goodwe inverters the comunication was decrypted but it seems that is not the case for the smart meter Homekit 1000 ([another conversation](https://forums.whirlpool.net.au/archive/90x1x50n) talking about that).

I have not found anything useful to connect locally to a Goodwe Homekit 1000 :(

**Reply by mfaust78** [reply]
Just my thoughts:
As the Home Kit (or more precisely the Smart Meter) has two RS485 connectors, you could try to connect via RS485 or a RS485/TCP bridge. Protocol is Modbus, 9600 Baud. Unfortunately GoodWe does not publish much information, but as the Home Kit or the Smart Meters can be used with other Inverters, there sure is some information which register is which value - but it will need some reverse engineering. If you figured out the registers, you could use HA modbus to read the registers. By far not as nice as the full integration as it exists for the inverters.

**Reply by icepicknz** [reply]
> Yes my installation is different. Meter is linked to the inverter only. Not connected on WiFi.
> 
> The meter then sends the data to the inverter.
> 
> This integrateion is for goodwe **inverters**
> 
> You'll need to look for another integration for the HK1000 meter separately. Or link the meter to the inverter and obtain the information from there.

My meter is connected to my home wifi, I can connect to the web interface of it and its similar setup in its panel. but I cannot collect any data dont know how. would be very useful to collect load from to grid and from house usage

**Comment by Hufflepuff-source** [comment]
I’m in the same situation. I can see the data from the homekit through the
SEMS portal but can’t seem to retrieve it and display it on home Assistant.

Also the GoodWe integration keeps saying unable to connect via the ip
address but I am able to retrieve data from the inverter through the SEMS
hacs api integration. ☹️



On Mon, 18 Jul 2022 at 6:31 am, icepicknz ***@***.***> wrote:

> Yes my installation is different. Meter is linked to the inverter only.
> Not connected on WiFi.
>
> The meter then sends the data to the inverter.
>
> This integrateion is for goodwe *inverters*
>
> You'll need to look for another integration for the HK1000 meter
> separately. Or link the meter to the inverter and obtain the information
> from there.
>
> My meter is connected to my home wifi, I can connect to the web interface
> of it and its similar setup in its panel. but I cannot collect any data
> dont know how. would be very useful to collect load from to grid and from
> house usage
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mletenay/home-assistant-goodwe-inverter/discussions/113#discussioncomment-3167024>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASK5K6ABUJMML6AWQDH7UQTVURUT5ANCNFSM5PEHTYDA>
> .
> You are receiving this because you commented.Message ID:
> <mletenay/home-assistant-goodwe-inverter/repo-discussions/113/comments/3167024
> @github.com>
>
-- 

Thanks and regards,


*SHIVSHANKAR MURALI*

+61 48177 3039

**Reply by mletenay** [reply]
Well, this is community project without any involvement of the inverter producer, so unless you (or someone with direct access to the device) will figure something out, there's not much to do.
Also, it is well possible that the HomeKit does not provide any network accessible interface (like inverter) at all.
The fact it reports to SEMS means nothing, the same applies to the inverter itself. 
It's their independent feature to regularly upload some runtime data (in encrypted form) to cloud.
The inverter exposes its modbus protocol (besides the RS485) also over IP/UDP, since that's the way it is managed by the mobile applications.
If the HomeKit does not have mobile app, it's very likely there's no modbus over IP at all.
In such case, your only options are to write completely new integration which will get/parse some data from HomeKit's html interfaces.

**Comment by Hufflepuff-source** [comment]
Hk1000 is a smart meter attached to your electricity meter to show how much
solar you are producing, energy consumption from solar and grid and how
much you are sending it back to the grid.

The inverter only shows you how much you are producing.



On Fri, 15 Jul 2022 at 11:11 pm, SniperHilton ***@***.***>
wrote:

> Isn't the hk1000 just the meter?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mletenay/home-assistant-goodwe-inverter/discussions/113#discussioncomment-3156647>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASK5K6E6FFIBUIWP3JRNEDLVUFPRBANCNFSM5PEHTYDA>
> .
> You are receiving this because you commented.Message ID:
> <mletenay/home-assistant-goodwe-inverter/repo-discussions/113/comments/3156647
> @github.com>
>
-- 

Thanks and regards,


*SHIVSHANKAR MURALI*

+61 48177 3039

**Comment by smlx** [comment]
Hey in case it helps I figured out how to get HK1000 metrics locally. Unfortunately it's done by reverse-engineering the SEMS Portal traffic, not UDP Modbus. So might not be a good fit for this project: https://github.com/smlx/goodwe-exporters

**Reply by ongas** [reply]
> Hey in case it helps I figured out how to get HK1000 metrics locally. Unfortunately it's done by reverse-engineering the SEMS Portal traffic, not UDP Modbus. So might not be a good fit for this project: https://github.com/smlx/goodwe-exporters

Hi smlx,

Thanks for all your work on this subject :)
I have been able to use your work as a starting point to develop a few integrations for Home Assistant and a HK3000!
The Integrations are in beta as I have been testing solo, but I have run each of them for several weeks in a live environment and they seem to work well.
If anyone is interested in getting more frequent data from a HK3000, and you're ok with a beta release, let me know.
